src/olympia/addons/serializers.py

```diff
@@ -1,33 +1,120 @@
+
+    # FIXME:
+    # - license
+    # - appversion compatibility info
+    # - release notes (separate endpoint ?)
+    # - all the reviewer/admin fields (different serializer/endpoint)
+
+    class Meta:
+        model = Version
+        fields = ('id', 'files', 'reviewed', 'version')
 
 
 class AddonSerializer(serializers.ModelSerializer):
-    name = TranslationSerializerField()
+    current_version = VersionSerializer()
     description = TranslationSerializerField()
+    homepage = TranslationSerializerField()
+    name = TranslationSerializerField()
+    status = serializers.Field(source='get_status_display')
+    summary = TranslationSerializerField()
+    support_email = TranslationSerializerField()
+    support_url = TranslationSerializerField()
+    tags = serializers.SerializerMethodField('get_tags')
+    type = serializers.Field(source='get_type_display')
+    url = serializers.SerializerMethodField('get_url')
+
+    # FIXME:
+    # - categories (need to sort out the id/slug mess in existing search code)
+    # - icon/previews
+    # - average rating, number of downloads, hotness
+    # - dictionary-specific things
+    # - persona-specific things
+    # - contributions-related things
+    # - annoying/thankyou and related fields
+    # - authors
+    # - dependencies, site_specific, external_software
+    # - thereason/thefuture (different endpoint ?)
+    # - in collections, other add-ons by author, eula, privacy policy
+    # - eula / privacy policy (different endpoint)
+    # - all the reviewer/admin-specific fields (different serializer/endpoint)
 
     class Meta:
         model = Addon
-        fields = ('id', 'default_locale', 'name', 'last_updated', 'slug')
+        fields = ('id', 'current_version', 'default_locale', 'description',
+                  'guid', 'homepage', 'name', 'last_updated', 'public_stats',
+                  'slug', 'status', 'summary', 'support_email', 'support_url',
+                  'tags', 'type', 'url')
+
+    def get_tags(self, obj):
+        if not hasattr(obj, 'tag_list'):
+            attach_tags([obj])
+        return getattr(obj, 'tag_list', [])
+
+    def get_url(self, obj):
+        return absolutify(obj.get_url_path())
 
 
 class ESAddonSerializer(BaseESSerializer, AddonSerializer):
     datetime_fields = ('last_updated',)
+    translated_fields = ('name', 'description', 'homepage', 'summary',
+                         'support_email', 'support_url')
 
     def fake_object(self, data):
         """Create a fake instance of Addon and related models from ES data."""
-        obj = Addon(id=data.id, slug=data.slug)
+        obj = Addon(id=data.id, slug=data.slug, is_listed=True)
 
-        # Set base attributes that have the same name/format in ES and in the
-        # model.
+        if data['current_version'] and data['current_version']['files']:
+            data_version = data['current_version']
+            obj._current_version = Version(
```

Comment:
It seems like this would be nicer as something like `obj._current_version = ESVersionSerializer().fake_object(data['current_version'])`. If we're never pulling versions out directly then it really doesn't matter though.